### PR TITLE
Adds localhost install feature (rebased version of PR #348)

### PR DESCRIPTION
**Pull request by prasadtalasila**

### CODE REVIEW

**Review by nichlaes — APPROVED** [review]

**Comment by prasadtalasila** [comment]
@nichlaes , one of the problems is that the default username is `user1`. This leads to problems for localhost and trial installation. The users are likely to use https://gitlab.com as authentication provider with user-level OAuth application. These users will have their own gitlab account. We need to make username as another argument to the single-script-install.sh. Can you please look into this? Thanks.

File: deploy/single-script-install.sh [hunk 1]
```diff
@@ -167,24 +194,16 @@ docker run -d \
...
-
 #-------------
 printf "\n\n Start the traefik gateway server\n "
 printf "...........\n "
 cd "${TOP_DIR}/servers/config/gateway" || exit
 cp "${TOP_DIR}/deploy/config/gateway/auth" auth
-cp "${TOP_DIR}/deploy/config/gateway/fileConfig.yml" "dynamic/fileConfig.yml"
-
+if [ -n "$env_variable" ] ; then
+  cp "${TOP_DIR}/deploy/config/gateway/fileConfig.${env_variable}.yml" "dynamic/fileConfig.yml"
```
Comment (prasadtalasila) [inline deploy/single-script-install.sh:203]: sed "s/user1/${username}/" "${TOP_DIR}/deploy/config/gateway/fileConfig.${env_variable}.yml" > "dynamic/fileConfig.yml"

File: deploy/single-script-install.sh [hunk 2]
```diff
@@ -167,24 +194,16 @@ docker run -d \
   --restart always \
```
Comment (prasadtalasila) [inline deploy/single-script-install.sh:194]:
```sh
cp -R "${TOP_DIR}/files/user1" "${TOP_DIR}/files/${username}"
docker run -d \
 -p 8090:8080 \
  --name "ml-workspace-${username}" \
  -v "${TOP_DIR}/files/user1:/workspace" \
  -v "${TOP_DIR}/files/common:/workspace/common" \
  --env AUTHENTICATE_VIA_JUPYTER="" \
  --env WORKSPACE_BASE_URL="${username}" \
  --shm-size 512m \
  --restart always \
  mltooling/ml-workspace-minimal:0.13.2 || true
```

File: docs/admin/localhost.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+!!! note
+    While installing you might encounter multiple dialogs asking,
+    which services should be restarted. Just click **OK** to all of those.
+
+Run the following scripts. To setup the installation to use localhost,
+the following argument is added to the script `--env local`.
+
+```bash
+wget https://raw.githubusercontent.com/INTO-CPS-Association/DTaaS/feature/distributed-demo/deploy/single-script-install.sh
+bash single-script-install.sh --env local
```
Comment (prasadtalasila) [inline docs/admin/localhost.md:57]: bash single-script-install.sh --env local --user <username>

File: docs/admin/trial.md [hunk 4]
```diff
@@ -49,11 +54,12 @@ Remember to create gitlab accounts for `user1` and `user2`.
...
     which services should be restarted. Just click **OK** to all of those.
 
-Run the following scripts.
+Run the following scripts. To setup the installation to use trial,
+the following argument is added to the script `--env trial`.
 
 ```bash
 wget https://raw.githubusercontent.com/INTO-CPS-Association/DTaaS/feature/distributed-demo/deploy/single-script-install.sh
-bash single-script-install.sh
+bash single-script-install.sh --env trial
```
Comment (prasadtalasila) [inline docs/admin/trial.md:62]: bash single-script-install.sh --env trial --user <username>

**Pull request merged**